ui/app/styles/core/icon.scss

```diff
@@ -0,0 +1,37 @@
+$icon-dimensions: 1.25rem;
+$icon-dimensions-small: 1rem;
+$icon-dimensions-medium: 1.5rem;
+$icon-dimensions-large: 2rem;
+
+.icon {
+  align-items: center;
+  display: inline-flex;
+  justify-content: center;
+  vertical-align: text-top;
+  height: $icon-dimensions;
+  width: $icon-dimensions;
+  fill: lighten($text, 25%);
+
+  &.is-small {
+    height: $icon-dimensions-small;
+    width: $icon-dimensions-small;
+  }
+
+  &.is-medium {
+    height: $icon-dimensions-medium;
+    width: $icon-dimensions-medium;
+  }
+
+  &.is-large {
+    height: $icon-dimensions-large;
+    width: $icon-dimensions-large;
+  }
+
+  @each $name, $pair in $colors {
+    $color: nth($pair, 1);
+
+    &.is-#{$name} {
+      fill: $color;
```

Comment:
Could get rid of the loop, and use `currentColor` here. You'd get rid of some CSS, but then you'd have to use the text helpers on the container or on the SVG to set the color and they're a tad more cumbersome imo sooo maybe 👌 : as it is 😄

Reply:
Interesting idea though! Maybe make it `fill: currentColor` with no modifiers, then allow modifiers for situations like this, where it's standalone.